vidarr-cli/src/main/java/ca/on/oicr/gsi/vidarr/cli/TestValidatorScript.java

```diff
@@ -19,15 +19,27 @@ public final class TestValidatorScript extends TestValidator {
   private String outputMetrics;
 
   @Override
-  public Validator createValidator() {
+  public Validator createValidator(String outputDirectory, String id, boolean verboseMode) {
     try {
-      final var tempDir = Files.createTempDirectory("vidarr-test-script");
-      final var calculateScript = tempDir.resolve("calculate");
-      final var calculateDir = tempDir.resolve("output");
-      Files.createDirectories(calculateDir);
-      Files.copy(Path.of(metricsCalculate), calculateScript);
+      // If output directory provided, we use that
+      // else we create a new directory in the default temporary-file directory
+      // Note: That path is associated with the default FileSystem which is UNIX in our case
+      final var finalDir = (outputDirectory != null) ? Path.of(outputDirectory)
+          : Files.createTempDirectory("vidarr-test-script");
+      final var finalCalculateScript =
+          (outputDirectory != null) ? finalDir.resolve(id) :
+              finalDir.resolve("calculate");
+      final var finalCalculateDir = finalDir.resolve("output");
+
+      // If outputDirectory provided then output file name will be: "id.output"
+      // Otherwise it will be: "calculate.output" if no output directory passed in
+      final var finalOutputFile = (outputDirectory != null) ? id + ".output" : "calculate.output";
```

Comment:
Although, also I guess, why? What do we get out of naming the file 'test1' or whatever instead of 'calculate'?

Reply:
Workflow developers are already used to `calculate.output`, this change is probably less clear